app/src/main/java/com/amsterdam/cutetudee/domain/exception/Exceptions.kt

```
@@ -1,2 +1,31 @@
 package com.amsterdam.cutetudee.domain.exception
 
+import kotlinx.datetime.LocalDate
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+
+open class CuteTudeeExceptions(message: String) : Exception(message)
+
+
+open class TaskExceptions(message: String) : CuteTudeeExceptions(message)
+
+@OptIn(ExperimentalUuidApi::class)
+class TaskNotFoundException(taskId: Uuid) : TaskExceptions("Task not found with id: $taskId")
```

We can make the exception take throwable instead of adding hardcoded message here.
already we known that this is not founded task.

---

what is the point of that .
the message here for debugging

---

For debugging we need the whole exception message not a custom message ... what if the error happends in the data base. By this you will not know just we will no its no task not the correct exception🤦‍♀️

---

no that is not correct. if there is exception in the data base then we will throw Custome exception that exist in domain and we will map the exceptions from our database framework to our domain